src/radical/pilot/agent/launch_method/mpirun.py

```diff
@@ -152,6 +152,10 @@ def get_launch_cmds(self, task, exec_path):
         td         = task['description']
         sandbox    = task['task_sandbox_path']
         task_cores = td.get('cores_per_rank', 1)
+        task_gpus  = td.get('gpus_per_rank', 0.)
+        # sharing GPUs among multiple ranks not supported
```

Comment:
Why can't we support this actually?  Ultimately it means that we would use the same GPU ID for multiple ranks - or is there more to it?  I don't think `mpirun` needs to be aware of how GPUs are allocated and used at all.

Reply:
that statement is to our LM, and not to `mpirun` itself, and that just a follow-up after our discussion, as we wanted to allow GPU sharing for JSRun LM only (at least for now)

Reply:
Fair.  I wouldn't mind reconsidering this though since it seems to fall out for free and we add an artificial limitation with this check.

Reply:
removed check for both MPIRun and SRun LMs, and since SRun uses actual number of GPUs per task/rank, then it takes such information from `slots` (and not from a corresponding task description)